### PR TITLE
Zoom Support utilities

### PR DESCRIPTION
**Pull request by Jonasdoubleyou**
Allow Admins to create and delete Zoom users and delete Zoom meetings, as well as query them.

### CODE REVIEW

File: common/zoom/zoom-scheduled-meeting.ts [hunk 1]
```diff
@@ -152,11 +154,12 @@ const deleteZoomMeeting = async (meetingId: string) => {
...
     );
 
-    if (response.status === 204) {
-        logger.info(`Zoom - The Zoom Meeting ${meetingId} was deleted.`);
-    } else {
+    if (!response.ok) {
         throw new Error(`Zoom - Failed to delete meeting with ${response.status} ${response.statusText}`);
     }
+
+    await prisma.lecture.update({ where: { id: appointment.id }, data: { zoomMeetingId: null } });
```
Comment (Jonasdoubleyou) [inline common/zoom/zoom-scheduled-meeting.ts:161]: I think this cleanup of the meetingId was missing before ...

File: common/zoom/zoom-user.ts [hunk 2]
```diff
@@ -197,12 +197,13 @@ const deleteZoomUser = async (zoomUserId: string) => {
...
 
-    if (response.status === 204) {
-        logger.info(`Zoom - Deleted Zoom user ${zoomUserId}`);
-    } else {
-        logger.error(`Zoom - ${response.statusText}`);
+    if (!response.ok) {
+        throw new Error(`Zoom failed to delete user for Student(${student.id}) ${response.statusText}`);
     }
 
+    await prisma.student.update({ where: { id: student.id }, data: { zoomUserId: null } });
```
Comment (Jonasdoubleyou) [inline common/zoom/zoom-user.ts:204]: same here

File: graphql/appointment/fields.ts [hunk 3]
```diff
@@ -151,12 +153,14 @@ export class ExtendedFieldsLectureResolver {
                 return appointment.title || 'Kein Titel';
         }
     }
-    @FieldResolver((returns) => String)
-    @Authorized(Role.USER)
-    zoomMeetingId(@Root() appointment: Appointment): string {
```
Comment (Jonasdoubleyou) [inline graphql/appointment/fields.ts:156]: There is already a default field resolver for the field (annotated with authorization checks for PARTICIPANT and OWNER). I'm not sure if every USER is supposed to know the MeetingId? Or doesn't it matter?

Comment (timidak) [reply graphql/appointment/fields.ts:156]: That default should be totally fine, I guess.

**Comment by timidak** [comment]
This issue tackles this issue: https://github.com/corona-school/project-user/issues/763
Thank you for your work! 🥇

**Review by timidak — APPROVED** [review]
Nice job! 👍🏻

**Pull request merged**